curp/src/server/raw_curp/mod.rs

```diff
@@ -812,20 +828,43 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
             .ctx
             .sp
             .map_lock(|sp| sp.pool.values().cloned().collect());
-        cst.votes_received = 1;
-        cst.sps = HashMap::from([(self.id().clone(), self_sp)]);
 
         if prev_role == Role::Follower {
             debug!("Follower {} starts election", self.id());
         } else {
             debug!("Candidate {} restarts election", self.id());
         }
 
-        Vote {
-            term: st.term,
-            candidate_id: self.id().clone(),
-            last_log_index: log.last_log_index(),
-            last_log_term: log.last_log_term(),
+        // vote to self
+        let election_ends = {
```

Comment:
This `election_ends` looks redundant.

```suggestion
        if cst.votes_received < min_granted {
            Some(Vote {
                term: st.term,
                candidate_id: self.id().clone(),
                last_log_index: log.last_log_index(),
                last_log_term: log.last_log_term(),
            })
        } else {
            // single node cluster
            // vote is granted by the majority of servers, can become leader
            let spec_pools = cst.sps.drain().collect();
            let mut log_w = RwLockUpgradableReadGuard::upgrade(log);
            self.recover_from_spec_pools(st, &mut log_w, spec_pools);
            self.become_leader(st);
            None
        }
```

Reply:
Fixed